src/models/caches.jl

```diff
@@ -22,3 +22,23 @@ struct PolarNetworkState{VT} <: AbstractNetworkBuffer
     dx::VT
 end
 
+function PolarNetworkState(state::PolarNetworkState, device = nothing)
+    if device == CUDADevice()
+        VT = CuVector{Float64}
+    elseif device == CPU()
+        VT = Vector{Float64}
+    else
+        VT = typeof(state.vmag)
```

Comment:
Here, I think we should find a better way to infer the type rather than using this hardcoded line (but I confess I am quite guilty of this in other parts of the code...)

Reply:
I think it would make sense to gather all comments porting on the type inference in a separate issue, that we could discuss together